casepro/msgs/models.py

```diff
@@ -356,7 +356,7 @@ def search(cls, org, user, search):
         all_label_access = user.can_administer(org)
 
         if all_label_access:
-            if folder == MessageFolder.inbox:
+            if folder == MessageFolder.inbox or (folder == MessageFolder.archived and label_id):
```

Comment:
Not getting why `and label_id` is here. Isn't this the path we want when an admin is viewing all archived messages?